### DISCUSSION TITLE
spark.submit.deployMode not work

### DISCUSSION THREAD

**Discussion by imperio-wxm** [apache/kyuubi#4679] [Q&A]
When spark.submit.deployMode=cluster is configured, the log shows that kyuubi's spark-submit submission command has taken effect, but the parameter on spark-web is still client

Log
```shell
2023-04-10 14:30:04.791  INFO 6 --- [engine-launch-log] o.a.kyuubi.jdbc.hive.KyuubiConnection    : 	--conf spark.submit.deployMode=cluster \
```

Spark-web
![image](https://user-images.githubusercontent.com/8111349/230843601-15e3d3d2-2a60-45fa-8c30-72ab8e9bdc89.png)

**Comment by pan3793** [comment]
> When spark.submit.deployMode=cluster is configured ...

How?

**Reply by imperio-wxm** [reply]
Set by jdbc sessionProperties, cpu, memory and other parameters are all right.
![image](https://user-images.githubusercontent.com/8111349/230844466-dc2cb9c1-d779-4d68-993d-74496b3f0d24.png)

**Reply by pan3793** [reply]
any other info? e.g. resource manager is k8s or yarn or something else?

**Reply by imperio-wxm** [reply]
resource manager is k8s and found that the client is specified in the official entrypoint.sh of spark. I don’t know if it has anything to do with this.
![image](https://user-images.githubusercontent.com/8111349/230845136-527563b0-f65a-43fc-9f5a-437a35493858.png)

**Reply by pan3793** [reply]
it's by design, the Spark on K8s cluster mode can be judged by the following two configurations.
```
spark.submit.deployMode=client
spark.kubernetes.submitInDriver=true
```

**Reply by pan3793** [reply]
For more implementation details, please read code `org.apache.spark.deploy.SparkSubmit`

**Reply by imperio-wxm** [reply]
spark.kubernetes.submitInDriver
Do you mean that only setting `spark.submit.deployMode=cluster` cannot take effect, need set `spark.kubernetes.submitInDriver=true`? But I can't find this parameter on spark official website: https://spark.apache.org/docs/latest/running-on-kubernetes.html

**Reply by pan3793** [reply]
No.

The user interface is unified, `spark-submit --conf spark.submit.deployMode=cluster` or `spark-submit --deploy-mode cluster` is sufficient. But the internal implementation is a little bit hacky in K8s cluster mode compared w/ others.

I mean, a running Spark application that runs on K8s, whose deploy mode can be judged by two configurations.

**Comment by yaooqinn** [comment]
It might be just a display issue, you can check whether there is a driver pod or not

**Reply by imperio-wxm** [reply]
The driver pod is available. Maybe it's a web display problem as you said
